### PR TITLE
6pm d add sorting to courses

### PR DESCRIPTION
**Pull request by willmori**
In this PR, we added ascending and descending sorting to various fields on the courses table.

Since this is done at the React Component for the CourseTable, the change shows up every place that component is used (both the Home page, and the page where Admin users do CRUD operations on Courses.)

Before:

proj-ucsb-cs-las-pr-62-before

After:

proj-ucsb-cs-las-pr-62-after

### CODE REVIEW

**Comment by andrewhlu** [comment]
I'm going to unassign myself from this review. Please request your team's designated LA and TA instead. Thanks!

Also, you might want to add the DS store files to the gitignore and remove them from the PR.

**Review by scottpchow23 — DISMISSED** [review]
This looks good but we should remove the `.DS_Store` files and then add the following line to the `.gitignore`.

```
**/.DS_Store
```

This will prevent `.DS_Store` files from being added to the repo in the future.

**Review by alejandrodiera — COMMENTED** [review]
Looked through and everything checks out. Good work!

File: javascript/src/main/components/Courses/CourseTable.js [hunk 1]
```diff
@@ -21,24 +21,51 @@ export default ({courses,admin,deleteCourse}) => {
...
+        const descendingON = String.fromCharCode(0x25bc);
+        const ascendingOFF = String.fromCharCode(0x25b3);
+        const descendingOFF = String.fromCharCode(0x25bd);
+
+        if (!order) 
+            return (<span data-testid="sort">&nbsp;&nbsp;{descendingOFF}{ascendingOFF}</span>);
+        else if (order === 'asc') 
+            return (<span data-testid="sort-asc">&nbsp;&nbsp;{descendingOFF}<font color="red">{ascendingON}</font></span>);
+        else 
+            return (<span data-testid="sort-desc">&nbsp;&nbsp;<font color="red">{descendingON}</font>{ascendingOFF}</span>);
```
Comment (scottpchow23) [inline javascript/src/main/components/Courses/CourseTable.js:35]: I'd remove the non-breaking spaces:
```suggestion
        if (!order) 
            return (<span data-testid="sort">{descendingOFF}{ascendingOFF}</span>);
        else if (order === 'asc') 
            return (<span data-testid="sort-asc">{descendingOFF}<font color="red">{ascendingON}</font></span>);
        else 
            return (<span data-testid="sort-desc"><font color="red">{descendingON}</font>{ascendingOFF}</span>);
```

File: javascript/src/test/components/Courses/CourseTable.test.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  });
+
+  test("ascending and descending buttons can be pressed", () => {
+    const { getByText } = render(<CourseTable courses={courseList} admin={true} deleteCourse={deleteCourse} />);
+    const courseNumberHeader = getByText(/Course Number/);
+
+    //cycle through both ascending and descending for test coverage
+    userEvent.click(courseNumberHeader);
+    userEvent.click(courseNumberHeader);
+
```
Comment (scottpchow23) [inline javascript/src/test/components/Courses/CourseTable.test.js:22]: Can we verify in between toggling the ascending and descending sort whether the _other_ version of each sort caret shows up? Something like this: 
```suggestion
const descendingOFF = String.fromCharCode(0x25bd);
expect(getByText(descendingOFF)).toBeInTheDocument();
```

**Review by scottpchow23 — CHANGES_REQUESTED** [review]
Couple of suggestions, nearly there!

**Review by pconrad — APPROVED** [review]
LGTM!

**Comment by pconrad** [comment]
40 points!

**Pull request merged**